### PR TITLE
Update install latest release to 0.5.0

### PR DESCRIPTION
**Pull request by manuelwallrapp**
# Changes

Point the install documentation to latest release 0.5.0

/kind documentation

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
/lgtm

**Review by khrm — COMMENTED** [review]
/ok-to-test

**Review by khrm — COMMENTED** [review]
/kind misc

**Comment by khrm** [comment]
@manuelwallrapp Please sign cla.

**Review by dibyom — APPROVED** [review]

File: docs/install.md [hunk 1]
```diff
@@ -49,7 +49,7 @@
 ## Installing latest release
 
 ```sh
-$ kubectl apply -f https://storage.googleapis.com/tekton-releases/results/previous/v0.3.0/release.yaml
+$ kubectl apply -f https://storage.googleapis.com/tekton-releases/results/previous/v0.5.0/release.yaml
```
Comment (enarha) [inline docs/install.md:52]: Now we have link to latest: https://storage.googleapis.com/tekton-releases/results/latest/release.yaml
Use it and we won't need to update this anymore.

**Review by avinal — APPROVED** [review]

**Review by khrm — COMMENTED** [review]
/lgtm

**Pull request merged**